addons/binding/org.openhab.binding.feican/src/main/java/org/openhab/binding/feican/handler/FeicanHandler.java

```diff
@@ -0,0 +1,204 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.feican.handler;
+
+import static org.openhab.binding.feican.FeicanBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.feican.internal.Commands;
+import org.openhab.binding.feican.internal.Connection;
+import org.openhab.binding.feican.internal.FeicanConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FeicanHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+public class FeicanHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FeicanHandler.class);
+    private final Commands commands = new Commands();
+
+    @Nullable
+    private Connection connection;
+
+    /**
+     * Creates a new handler for the Feican thing.
+     *
+     * @param thing The thing to create the handler for
+     */
+    public FeicanHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof OnOffType) {
+                handleSwithOnOff((OnOffType) command);
+            } else if (command instanceof HSBType) {
+                handleColor(channelUID, (HSBType) command);
+            } else if (command instanceof PercentType) {
+                handlePercentage(channelUID, (PercentType) command);
+            } else if (command instanceof StringType) {
+                handleString(channelUID, (StringType) command);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final FeicanConfiguration config = getConfigAs(FeicanConfiguration.class);
+
+        logger.info("Initializing Feican Wifi RGWB Bulb on IP address {}", config.ipAddress);
+        connection = new Connection(config.ipAddress);
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
Do you also get here if the connection fails? Then it should not be set to ONLINE.

Reply:
The Connection constructor doesn't make a connection. Because it doesn't support getting the status. It's only really known to be not online when the user sends a command. So here it's just assumed the status is online. Otherwise it can only be shown only after the user has send a command. Or one solution could be to see if sending some dummy command to the device gives a different result for a device that is offline and one that is online. This as a kind of ping to check if the device is online.

Reply:
I see, thanks.